packages/pipeline/src/utils/getPipelineCounterTooltip.ts

```diff
@@ -0,0 +1,28 @@
+import { ReactNode } from 'react';
+import { onlyText } from 'react-children-utilities';
+import { flatMap } from 'lodash';
+import { flow, compact, split, map } from 'lodash/fp';
+
+/**
+ * A utility function which takes the React.children rendered by the Pipeline component
+ * and returns the string which represents the full pipeline, for usage with the tooltip.
+ *
+ * @param children - the children rendered by the pipeline component
+ * @returns string - the tooltip text
+ */
+export function getPipelineCounterTooltip(children: ReactNode): string {
+  const stages = flow(
+    onlyText,
+    split('$'),
+    compact,
+    map(str => `$${str}`),
+  )(children);
+
+  const formattedStages = flatMap(stages, (value, index, array) =>
```

Comment:
I think our build supports flatMap as an array method (https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/flatMap).

Reply:
If not, for tree-shaking purposes, we should import flatMap like so:
```
import flatMap from 'lodash/flatMap'
```

Reply:
Resolved by ~~fc2aade~~ 52053b1